Frameworks/MobileCoreServices/UTType.mm

```diff
@@ -14,134 +14,137 @@
+const CFStringRef kUTTypeMPEG = CFSTR("public.mpeg");
+const CFStringRef kUTTypeMPEG4 = CFSTR("public.mpeg-4");
+const CFStringRef kUTTypeMP3 = CFSTR("public.mp");
+const CFStringRef kUTTypeMPEG4Audio = CFSTR("public.mpeg-4-audio");
+const CFStringRef kUTTypeAppleProtectedMPEG4Audio = CFSTR("com.apple.protected-mpeg-4-audio");
 
 // UTI Directory Types
-const CFStringRef kUTTypeFolder = static_cast<const CFStringRef>(@"public.folder");
-const CFStringRef kUTTypeVolume = static_cast<const CFStringRef>(@"public.volume");
-const CFStringRef kUTTypePackage = static_cast<const CFStringRef>(@"com.apple.package");
-const CFStringRef kUTTypeBundle = static_cast<const CFStringRef>(@"com.apple.bundle");
-const CFStringRef kUTTypeFramework = static_cast<const CFStringRef>(@"com.apple.framework");
+const CFStringRef kUTTypeFolder = CFSTR("public.folder");
+const CFStringRef kUTTypeVolume = CFSTR("public.volume");
+const CFStringRef kUTTypePackage = CFSTR("com.apple.package");
+const CFStringRef kUTTypeBundle = CFSTR("com.apple.bundle");
+const CFStringRef kUTTypeFramework = CFSTR("com.apple.framework");
 
 // UTI Miscellaneous Types
-const CFStringRef kUTTypeApplicationBundle = static_cast<const CFStringRef>(@"com.apple.application-bundle");
-const CFStringRef kUTTypeApplicationFile = static_cast<const CFStringRef>(@"com.apple.application-file");
-const CFStringRef kUTTypeVCard = static_cast<const CFStringRef>(@"public.vcard");
-const CFStringRef kUTTypeInkText = static_cast<const CFStringRef>(@"com.apple.ink.inktext");
+const CFStringRef kUTTypeApplicationBundle = CFSTR("com.apple.application-bundle");
+const CFStringRef kUTTypeApplicationFile = CFSTR("com.apple.application-file");
+const CFStringRef kUTTypeVCard = CFSTR("public.vcard");
+const CFStringRef kUTTypeInkText = CFSTR("com.apple.ink.inktext");
 
 /**
  @Status Caveat
  @Notes Only a subset of System UTIs are supported and inConformingToUTI is option not supported.
 */
 CFStringRef UTTypeCreatePreferredIdentifierForTag(CFStringRef inTagClass, CFStringRef inTag, CFStringRef inConformingToUTI) {
-    // Initialize MobileCoreServices System UTI.
+    RETURN_NULL_IF(!inTagClass);
+    RETURN_NULL_IF(!inTag);
+
+    // Initialize System UTI.
     _UTInitializeSystemUTIMaps();
 
     if (inConformingToUTI != NULL) {
         UNIMPLEMENTED_WITH_MSG("inConformingToUTI (%s) is not currently supported for UTTypeCreatePreferredIdentifierForTag",
-                               [static_cast<NSString*>(inConformingToUTI) UTF8String]);
+                               CFStringGetCStringPtr(inConformingToUTI, kCFStringEncodingUTF8));
     }
 
     CFComparisonResult result;
 
     result = CFStringCompare(inTagClass, kUTTagClassFilenameExtension, 0);
     if (result == kCFCompareEqualTo) {
-        CFArrayRef fileNameExtensions = _UTGetUTIsForFileNameExtension(inTag);
+        CFArrayRef fileNameExtensions = _UTCreateUTIsForFileNameExtension(inTag);
```

Comment:
on point. I had to fix these since as you mentioned most of the memory management here was not well written before.

---
In reply to: [95506147](https://github.com/Microsoft/WinObjC/pull/1662#discussion_r95506147) [](ancestors = 95506147)